src/formatDate.js

```diff
@@ -8,7 +8,38 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[3];
+  const newSeparator = toFormat[3];
+  const arrayFromDate = date.split(oldSeparator);
+  const result = [];
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    for (let j = 0; j < fromFormat.length; j++) {
+      if (toFormat[i] === 'YYYY' && fromFormat[j] === 'YY') {
+        fromFormat[j] = 'YYYY';
+
+        if (arrayFromDate[j] < 30) {
+          arrayFromDate[j] = '20' + arrayFromDate[j];
+        } else {
+          arrayFromDate[j] = '19' + arrayFromDate[j];
+        }
+      }
+
+      if (toFormat[i] === 'YY' && fromFormat[j] === 'YYYY') {
+        fromFormat[j] = 'YY';
+
+        const newFormatYear = arrayFromDate[j].slice(2);
+
+        arrayFromDate[j] = newFormatYear;
+      }
```

Comment:
Similar to the previous comment, modifying `fromFormat[j]` directly in the condition `toFormat[i] === 'YY' && fromFormat[j] === 'YYYY'` can cause issues. Use a temporary variable to store the converted format instead of changing the `fromFormat` array.